src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -4901,39 +4899,40 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
         argOffsetMax = compiler->lvaOutgoingArgSpaceSize;
     }
 
-    bool isStruct = (targetType == TYP_STRUCT) || (source->OperGet() == GT_FIELD_LIST);
+    GenTree* source = treeNode->gtGetOp1();
 
-    if (!isStruct) // a normal non-Struct argument
+    if (!source->TypeIs(TYP_STRUCT)) // a normal non-Struct argument
     {
-        if (varTypeIsSIMD(targetType))
+        if (varTypeIsSIMD(source->TypeGet()))
         {
             NYI_RISCV64("SIMD in genPutArgStk-----unimplemented/unused on RISCV64 yet----");
         }
 
-        instruction storeIns  = ins_Store(targetType);
-        emitAttr    storeAttr = emitTypeSize(targetType);
+        var_types   slotType  = genActualType(source);
+        instruction storeIns  = ins_Store(slotType);
+        emitAttr    storeAttr = emitTypeSize(slotType);
 
         // If it is contained then source must be the integer constant zero
         if (source->isContained())
         {
             assert(source->OperGet() == GT_CNS_INT);
             assert(source->AsIntConCommon()->IconValue() == 0);
-
-            emit->emitIns_S_R(storeIns, storeAttr, REG_R0, varNumOut, argOffsetOut);
-        }
-        else
-        {
-            genConsumeReg(source);
             if (storeIns == INS_sw)
             {
                 emit->emitIns_R_R_R(INS_addw, EA_4BYTE, source->GetRegNum(), source->GetRegNum(), REG_R0);
```

Comment:
Hm, what's the me meaning of  `addw source, source, R0`?

Reply:
I put `addw source, source, R0` to convert value of source into 4BYTE in case of `INS_sw`. Actually, I want to put the codes in the `else block`. However, I think it doesn't need. I removed. Thank you!